crates/miden-objects/src/note/note_tag.rs

```diff
@@ -248,23 +276,48 @@ impl fmt::Display for NoteTag {
 
 impl From<u32> for NoteTag {
     fn from(value: u32) -> Self {
-        Self(value)
+        // Mask out the two most significant bits.
+        match value & 0xc0000000 {
+            NETWORK_ACCOUNT => {
+                // Network, Specific Account (lower 30 bits are the account ID)
```

Comment:
Nit: I would not use the "Network, Specific" or "Local, Any" language here (I've changed that in the table I suggested in another comment).
I don't think these comments are necessary here, I think we can just remove them?